### PR TITLE
Add: Icarus-Dedicated-Server

### PR DESCRIPTION
**Pull request by BolverBlitz**
# Description

Adds the current Icarus Dedicated Server (Beta) to run in pterodactyl via wine.

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [X] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [X] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?

### CODE REVIEW

File: game_eggs/steamcmd_servers/icarus/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+### Authors / Contributors
+
+<!-- prettier-ignore-start -->
+<!-- markdownlint-disable -->
+<table>
+    <tr>
+        <td align="center">
+            <a href="https://github.com/BolverBlitz">
+                <img src="https://avatars.githubusercontent.com/u/35345288" width="50px;" alt=""/><br /><sub><b>Red-Thirten</b></sub>
```
Comment (parkervcp) [inline game_eggs/steamcmd_servers/icarus/README.md:14]: This still points at red-thirteens users FYI

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/icarus/egg-icarus--dedicated.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
...credist\r\nENV WINEARCH=win32\r\nENV WINEPATH=\/\r\n\r\n# Install packages. Default packages below are skipped if using image noted above, thus speeding up the install process.\r\napt -y update\r\npt -y --no-install-recommends install curl lib32gcc-s1 ca-certificates\r\n\r\n# Download and install SteamCMD\r\nexport HOME=\/mnt\/server\r\ncd \/tmp\r\nmkdir -p $HOME\/steamcmd $HOME\/steamapps\r\ncurl -sSL -o steamcmd.tar.gz https:\/\/steamcdn-a.akamaihd.net\/client\/installer\/steamcmd_linux.tar.gz\r\ntar -xzvf steamcmd.tar.gz -C $HOME\/steamcmd\r\ncd $HOME\/steamcmd\r\n\r\n# SteamCMD fails otherwise for some reason, even running as root.\r\n# This is changed at the end of the install process anyways.\r\nchown -R root:root \/mnt\r\n\r\n# Install game server using SteamCMD\r\n.\/steamcmd.sh +force_install_dir $HOME +login anonymous $( [[ \"${WINDOWS_INSTALL}\" == \"1\" ]] && printf %s '+@sSteamCmdForcePlatformType windows' ) +app_update 2089300 validate +quit\r\n\r\n# Set up 32 and 64 bit libraries\r\nmkdir -p $HOME\/.steam\/sdk{32,64}\r\ncp -v linux32\/steamclient.so $HOME\/.steam\/sdk32\/steamclient.so\r\ncp -v linux64\/steamclient.so $HOME\/.steam\/sdk64\/steamclient.so\r\n\r\nEOF",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/icarus/egg-icarus--dedicated.json:28]: 1. you use the installer image so the apt update and apt install on line 12 and 13 (even a `a` is missing on line 13)
2. wy is there just a random EOF at the end of the install script ?
3. where is the install done message that present in your example scripts?
4. this: `export HOME=/mnt/server` is missing on the line below `chown -R root:root /mnt` <https://github.com/parkervcp/eggs/blob/master/scripts/steamcmd_installer.sh>
5. wine is not even in the install image is this realy needed? : 
```bash
ENV WINEPREFIX=/home/steam/vcredist
ENV WINEARCH=win32
ENV WINEPATH=/
```

File: game_eggs/steamcmd_servers/icarus/egg-icarus--dedicated.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "WINETRICKS_RUN",
+            "description": "Run installs on software that is required by the server",
+            "env_variable": "WINETRICKS_RUN",
+            "default_value": "vcrun2019",
+            "user_viewable": false,
+            "user_editable": false,
+            "rules": "required|string",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/icarus/egg-icarus--dedicated.json:71]: if this is needed to start the server then this should be `required|string|in:vcrun2019`

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

**Comment by QuintenQVD0** [comment]
@BolverBlitz the apt update and apt install line are still there

**Comment by BolverBlitz** [comment]
Ok, sorry. I didn´t read that command.

**Comment by QuintenQVD0** [comment]
and a other note, the app id should be in a variable as that variable is used to auto update the server. for that read the comment on the first lines of the steamcmd install script

File: game_eggs/steamcmd_servers/icarus/egg-icarus--dedicated.json [hunk 4]
```diff
@@ -70,6 +70,26 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "AUTO_UPDATE",
+            "description": "Disabling or enabling automated updates on boot",
+            "env_variable": "AUTO_UPDATE",
+            "default_value": "1",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|in:0,1",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/icarus/egg-icarus--dedicated.json:91]: almost, this is the last thing that has to be changed: boolean

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

**Comment by gOOvER** [comment]
The DefaultGame.ini is Not shipped with the Server. you need to pull it from their repo

**Comment by BolverBlitz** [comment]
@gOOvER I can´t find this file on the discord or the the [Github](https://github.com/RocketWerkz/IcarusDedicatedServer) so i don´t know what i´m supposed to do.

**Comment by QuintenQVD0** [comment]
@BolverBlitz It looks like as goover suggested, the ServerSettings.ini must be pulled from somewhere or added to this pr and let the egg install script pull it from here. It also looks like -UserDir= is required because the default save game is here `%localappdata%\Icarus\` Maybe wine will Handel that I have no idea

**Comment by BolverBlitz** [comment]
@QuintenQVD0 The ServerSettings.ini file should be generated. And all my test deploys do have it.
grafik

You might be correct with the save game, however in the current state it does save all placed down things in a world (Restart persistent) 
Looking into my files, its stored in `/Icarus/Saved/PlayerData/DedicatedServer/Prospects/xxx.json`

**Comment by QuintenQVD0** [comment]
If it auto generates that config and auto saves to a path in /home/container then it is fine that way, then no modfications are needed

**Comment by QuintenQVD0** [comment]
If you can just provide a full startup log, then It is fine for me, Then I will merge it

**Comment by BolverBlitz** [comment]
https://nc.ebg.pw/fetch/6ffb7321log091222

**Comment by QuintenQVD0** [comment]
> https://nc.ebg.pw/fetch/6ffb7321log091222

what is this at the end?

```[2022.12.09-16.58.12:278][  1]IcarusOSSLog: Error: OnResUserTicket : No player found
0258:fixme:advapi:ReportEventW (00000000CAFE4242,0x0001,0x0000,0xc0000001,0000000000000000,0x0001,0x00000000,0000000000B9CC88,0000000000000000): stub
0258:err:eventlog:ReportEventW L"OnResUserTicket : No player found"
0298:fixme:cryptnet:check_ocsp_response_info check responder id
```

**Comment by BolverBlitz** [comment]
I don´t know exactly. Looking it up on the discord it seems perfectly fine for now.
I tested every commit by joining, building, saving, restarting and verifying the stuff is there.
grafik

**Review by QuintenQVD0 — APPROVED** [review]

**Review by QuintenQVD0 — APPROVED** [review]

**Review by parkervcp — APPROVED** [review]

**Comment by gOOvER** [comment]
> @BolverBlitz It looks like as goover suggested, the ServerSettings.ini must be pulled from somewhere or added to this pr and let the egg install script pull it from here. It also looks like -UserDir= is required because the default save game is here `%localappdata%\Icarus\` Maybe wine will Handel that I have no idea 

userdir is Not needed and make Problems.

i will post more suggetions tomorrow..i have nightshitft and must Work atm

**Pull request merged**